### PR TITLE
Add support for UGW 8.x (and OpenWRT in general)

### PR DESCRIPTION
**Pull request by arnout**
This PR replaces #497.

It is based on #511, adds a few more cleanup commits, and then the remaining commits from #497.

### CODE REVIEW

**Comment by arnout** [comment]
## ***This is not ready to merge yet***

I didn't have time to completely remove BEEROCKS_UGW and a few other things, or to think about a better way to avoid TARGET_PLATFORM_TYPE. I should have created this as a draft but accidentally didn't, and it's not possible to go back to draft :-(

File: common/beerocks/scripts/CMakeLists.txt [hunk 1]
```diff
@@ -3,8 +3,11 @@ if (TARGET_PLATFORM STREQUAL "linux")
     set(PLATFORM_DB_PATH ${CMAKE_INSTALL_PREFIX}/share)
     set(PLATFORM_DB_PATH_TEMP /tmp)
 elseif (TARGET_PLATFORM STREQUAL "openwrt")
-    set(PLATFORM_DB_PATH "/opt/prplmesh/share")
+    set(PLATFORM_DB_PATH "/opt/prplMesh/share")
     set(PLATFORM_DB_PATH_TEMP "/tmp/beerocks")
+    if (TARGET_PLATFORM_TYPE STREQUAL "ugw")
```
Comment (tomereli) [inline common/beerocks/scripts/CMakeLists.txt]: Not needed anymore

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 2]
```diff
@@ -265,6 +267,6 @@ DATA_IFACE=enx00ee22aa28ef
 PRPLMESH_MODE="CA" # CA = Controller & Agent, A = Agent only, C = Controller only
 
 # Export MultiAP libs folder
-export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:@INSTALL_PATH@/lib
+export LD_LIBRARY_PATH=@INSTALL_PATH@/lib:$LD_LIBRARY_PATH:@EXTRA_LD_PATH@
```
Comment (tomereli) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: `@EXTRA_LD_PATH@` not needed

File: controller/CMakeLists.txt [hunk 3]
```diff
@@ -5,13 +5,17 @@ set(BEEROCKS_WLAN1_IFACE "wlan0")
 set(BEEROCKS_WLAN2_IFACE "wlan2")
 
 # Platform specific flags
-if (TARGET_PLATFORM STREQUAL "ugw")
-    add_definitions(-DBEEROCKS_UGW)
+if (TARGET_PLATFORM STREQUAL "openwrt")
+    # UGW
+    if (TARGET_PLATFORM_TYPE STREQUAL "ugw")
+        add_definitions(-DBEEROCKS_UGW)
```
Comment (tomereli) [inline controller/CMakeLists.txt]: shouldn't be needed anymore

File: controller/CMakeLists.txt [hunk 4]
```diff
@@ -5,13 +5,17 @@ set(BEEROCKS_WLAN1_IFACE "wlan0")
...
+    # Turris Omnia
+    elseif (TARGET_PLATFORM_TYPE STREQUAL "turris-omnia")
+        set(BEEROCKS_WLAN2_IFACE "wlan1")
```
Comment (tomereli) [inline controller/CMakeLists.txt]: Did we come to a conclusion about this? I suggest we move the interface names to the platform DB, but that of course can be done in a follow-up PR.

**Comment by tomereli** [comment]
> ## _**This is not ready to merge yet**_
> I didn't have time to completely remove BEEROCKS_UGW and a few other things, or to think about a better way to avoid TARGET_PLATFORM_TYPE. I should have created this as a draft but accidentally didn't, and it's not possible to go back to draft :-(

Regarding `TARGET_PLATFORM_TYPE` - I don't think we need it at all **eventually** - `TARGET_PLATFORM` (we should probably change the name since it is more of a `TARGET_OS`) can be `openwrt`, `rdkb` or `linux`, but as of now I also don't see a way to remove it completely.
The usage of `TARGET_PLATFORM_TYPE` currently is for:

- scripts - add `/opt/intel/lib` to the linker search path - **not needed anymore**
- controller
  - Turris Omnia - set `BEEROCKS_WLAN2_IFACE` to be wlan1. Should be in platform DB so won't be needed **but in a different PR**
  - `-DBEEROCKS_UGW`
    - used for versioning. Some of it may be generic but **for a different PR**
    - UGW and RDKB use different safec libraries, so it is used in BWL (DWPAL) and BPL (UCI) to differentiate between them. We should require safec library used by UGW always and update our code 
to use safec anyway, and make a generic mechanism for including the correct headers.
    - agent - different configuration options for the conf files. Should be moved to platform DB
  - `-DBEEROCKS_RDKB` - similar to BEEROCKS_UGW


Bottom line, I think we should keep `TARGET_PLATFORM_TYPE` for now, but remove any dependency which is not mandatory. We should definitely create a task to move all non conventional platform related stuff to the platform DB.

File: tools/docker/builder/OpenWRT.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+# Platform specific arguments
+ARG PRPLMESH_PLATFORM_TYPE
+ARG PRPLMESH_PLATFORM_BASE_DIR
+ARG PRPLMESH_PLATFORM_BUILD_NAME
+ARG PRPLMESH_PLATFORM_TOOLCHAIN
+ARG PRPLMESH_PLATFORM_TOOLCHAIN_PREFIX
+
+# Make sure that all build arguments are supplied
+RUN  test -n "$PRPLMESH_PLATFORM_TYPE" \
```
Comment (tomereli) [inline tools/docker/builder/OpenWRT.Dockerfile:38]: Need to be removed. I don't have docker in the UGW build server..

File: tools/cmake/toolchain/openwrt.cmake [hunk 6]
```diff
@@ -0,0 +1,51 @@
+## General configuration
```
Comment (tomereli) [inline tools/cmake/toolchain/openwrt.cmake:1]: Missing license header

File: tools/cmake/toolchain/openwrt.cmake [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+# order which breaks this usage.
+#   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70129
+#   https://gitlab.kitware.com/cmake/cmake/issues/16291
+#   https://gitlab.kitware.com/cmake/cmake/issues/16919
+set(CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES ${PLATFORM_TOOLCHAIN_DIR} CACHE STRING "")
+set(CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES ${PLATFORM_TOOLCHAIN_DIR} CACHE STRING "")
+
+## Toolchain Programs
+set(CMAKE_C_COMPILER ${PLATFORM_TOOLCHAIN_DIR}/bin/${PLATFORM_TOOLCHAIN_PREFIX}gcc CACHE STRING "C Compiler")
+set(CMAKE_CXX_COMPILER ${PLATFORM_TOOLCHAIN_DIR}/bin/${PLATFORM_TOOLCHAIN_PREFIX}g++ CACHE STRING "C++ Compiler")
```
Comment (tomereli) [inline tools/cmake/toolchain/openwrt.cmake]: Missing line break

File: tools/cmake/toolchain/openwrt.cmake (hunk 6 above)
Comment (tomereli) [reply tools/cmake/toolchain/openwrt.cmake:1]: Actually it is not set for .cmake files, so skip it.

File: tools/docker/builder/OpenWRT.Dockerfile (hunk 5 above)
Comment (tomereli) [reply tools/docker/builder/OpenWRT.Dockerfile:38]: Will be fixed in a different PR.

**Review by tomereli — APPROVED** [review]
Fixed a bunch of issues UGW issues and an issue causing test_flows.sh to fail due to wrong platform db installation path linux after TMP_PATH changes.
Now I don't see a reason not to merge this so we have an official tag to use for the UGW prplmesh feed Makefile which I also prepared and tested, so approving for merge.
- other UGW issues
